### PR TITLE
psbt: implement BIP 371 Taproot PSBT fields

### PR DESCRIPTION
**Pull request by achow101**
Adds the Taproot PSBT fields as described in BIP 371

### CODE REVIEW

File: hwilib/psbt.py [hunk 1]
```diff
@@ -86,6 +88,12 @@ def __init__(self) -> None:
         self.hd_keypaths: Dict[bytes, KeyOriginInfo] = {}
         self.final_script_sig = b""
         self.final_script_witness = CTxInWitness()
+        self.tap_key_sig = b""
+        self.tap_script_sigs: Dict[Tuple[bytes, bytes], bytes] = {}
+        self.tap_scripts: Dict[Tuple[bytes, int], Set[bytes]] = {}
+        self.tap_keypaths: Dict[bytes, Tuple[Set[bytes], KeyOriginInfo]] = {}
```
Comment (Sjors) [inline hwilib/psbt.py:94]: 42963840ebe71e8553681e5ce80b84de8ba939ec : Same suggestion as on the Bitcoin Core side, to call this `tap_bip32_paths` or some other term that's not ambigious against keypath spend.

File: hwilib/psbt.py [hunk 2]
```diff
@@ -185,6 +199,72 @@ def deserialize(self, f: Readable) -> None:
                     raise PSBTSerializationError("final scriptWitness key is more than one byte type")
                 witness_bytes = BufferedReader(BytesIO(deser_string(f))) # type: ignore
                 self.final_script_witness.deserialize(witness_bytes)
+            elif key_type == 0x13:
```
Comment (Sjors) [inline hwilib/psbt.py:202]: 42963840ebe71e8553681e5ce80b84de8ba939ec : Maybe define these keytypes as constants in one place?

**Review by Sjors — APPROVED** [review]
utACK 171a5fb

File: hwilib/psbt.py (hunk 1 above)
Comment (achow101) [reply hwilib/psbt.py]: Done

**Comment by Sjors** [comment]
re-utACK c5fba03bd33a361d9ee488388d2e90cf73e61622

**Pull request merged**